test_tools/sigma_verify_rules.py

```diff
@@ -119,51 +56,61 @@ def run_verifier(rules_path, config_file_path):
         raise IOError('Config file path not found at path: {0:s}'.format(
             config_file_path))
 
-    sigma_config_path = config_file_path
+    sigma_config = sigma_util.get_sigma_config_file(
+        config_file=config_file_path)
 
-    with open(sigma_config_path, 'r') as sigma_config_file:
-        sigma_config_con = sigma_config_file.read()
-    sigma_config = sigma_configuration.SigmaConfiguration(sigma_config_con)
-    backend = sigma_elasticsearch.ElasticsearchQuerystringBackend(
-        sigma_config, {})
     return_verified_rules = []
     return_rules_with_problems = []
 
     for dirpath, dirnames, files in os.walk(rules_path):
-
         if 'deprecated' in [x.lower() for x in dirnames]:
             dirnames.remove('deprecated')
-            logger.info('deprecated in folder / filename found - ignored')
 
         for rule_filename in files:
-            if not rule_filename.lower().endswith(RULE_EXTENSIONS):
-                continue
+            if rule_filename.lower().endswith('.yml'):
+                # if a sub dir is found, do not try to parse it.
+                if os.path.isdir(os.path.join(dirpath, rule_filename)):
+                    continue
+
+                rule_file_path = os.path.join(dirpath, rule_filename)
+                parsed_rule = sigma_util.get_sigma_rule(
+                    rule_file_path, sigma_config)
+                if parsed_rule:
+                    return_verified_rules.append(rule_file_path)
+                else:
+                    return_rules_with_problems.append(rule_file_path)
 
-            # If a sub dir is found, skip it
-            if os.path.isdir(os.path.join(rules_path, rule_filename)):
-                logger.debug(
-                    'Directory found, skipping: {0:s}'.format(rule_filename))
-                continue
+    return return_verified_rules, return_rules_with_problems
 
-            rule_file_path = os.path.join(dirpath, rule_filename)
-            rule_file_path = os.path.abspath(rule_file_path)
 
-            if verify_rules_file(rule_file_path, sigma_config, backend):
-                return_verified_rules.append(rule_file_path)
-            else:
-                logger.info('File did not work{0:s}'.format(rule_file_path))
-                return_rules_with_problems.append(rule_file_path)
+def move_problematic_rule(filepath, move_to_path, reason=""):
+    """ Moves a problematic rule to a subfolder so it is not used again
+
+    Args:
+        filepath: path to the sigma rule that caused problems
+        move_to_path: path to move the problematic rules to
+        reason (optional): reason why file is moved
+
+    Returns:
+        Nothing
+    """
+
+    logging.info('Moving the rule: {0:s} to {1:s}'.format(
+        filepath, move_to_path))
+
+    os.makedirs(move_to_path, exist_ok=True)
+    file_object = open('{0:s}debug.log'.format(move_to_path), 'a')
```

Comment:
I would rather use `os.path.join` here ...

```
debug_path = os.path.join(move_to_path, 'debug.log')
```

and also instead of using open/close, can you use `with`

```
with open(debug_path, 'a') as file_objec:
  file_object.write...

```

since there are guarantees that the file handle is properly closed when you use the `with` syntax, which are not provided by doing it this way

Reply:
as always, clever, yes!